docs/hsi.xml

```diff
@@ -128,6 +128,13 @@
         connected, additional software to be installed, or disabling any existing
         security layers to measure.
       </para>
+      <para>
+        The HSI specification is primarily designed for consumer laptop, desktop
+        and server hardware, although some tests <emphasis>may</emphasis> still
+        make sense on embedded hardware such as mobile phones and other consumer
+        appliances.
+        It is not expected that embedded hardware will publish an HSI number.
```

Comment:
I think it's worth taking this a step further.  Have an allowlist of chassis types, and only calculate HSI for those.

Reply:
I'm not so sure the DMI data for chassis types is always correct; there's a lot of `To Be Filled By Te O.E.M.` out there.

Reply:
The manufacturer field sure, but the chasssis type?  Isn't that a number?
To me that's not a reason to ignore a spec though.  If the chassis type is bad information (isn't in the allow list), HSI can't be computed and show an error.

Reply:
> To me that's not a reason to ignore a spec though.

That's a fair enough point, but the devil is in the detail. Could you say which of these are "in" or "out" from an HSI point of view?

    02h Unknown
    03h Desktop
    04h Low Profile Desktop
    05h Pizza Box
    06h Mini Tower
    07h Tower
    08h Portable
    09h LapTop
    0Ah Notebook
    0Bh Hand Held
    0Ch Docking Station
    0Dh All in One
    0Eh Sub Notebook
    0Fh Space-saving
    10h Lunch Box
    11h Main Server Chassis
    12h Expansion Chassis
    13h SubChassis
    14h Bus Expansion Chassis
    15h Peripheral Chassis
    16h RAID Chassis
    17h Rack Mount Chassis
    18h Sealed-case PC
    19h Multi-system chassis
    1Ah Compact PCI
    1Bh Advanced TCA
    1Ch Blade
    1Dh Blade Enclosure

Reply:
I would say in is:
3, 4, 6, 7, 8, 9, A, D

Reply:
Filed as https://github.com/fwupd/fwupd/issues/2440